modules/apps/change-tracking/change-tracking-test/src/testIntegration/java/com/liferay/change/tracking/internal/portlet/action/test/AutocompleteUserMVCResourceCommandTest.java

```diff
@@ -0,0 +1,169 @@
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+
+import java.io.ByteArrayOutputStream;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Brooke Dalton
+ */
+@RunWith(Arquillian.class)
+public class AutocompleteUserMVCResourceCommandTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Before
+	public void setUp() throws Exception {
+		_user1 = UserTestUtil.addUser();
+		_user2 = UserTestUtil.addUser();
+
+		_permissionChecker = PermissionCheckerFactoryUtil.create(_user1);
+
+		UserGroup userGroup = UserGroupTestUtil.addUserGroup();
+
+		Role role = _roleLocalService.getRole(
+			TestPropsValues.getCompanyId(), RoleConstants.PUBLICATIONS_USER);
+
+		RoleTestUtil.addResourcePermission(
+			role, CTCollection.class.getName(), ResourceConstants.SCOPE_COMPANY,
+			String.valueOf(_user1.getCompanyId()), CTActionKeys.INVITE_USERS);
+
+		_groupLocalService.addRoleGroup(
+			role.getRoleId(), userGroup.getGroupId());
+
+		_userLocalService.addUserGroupUser(
+			userGroup.getUserGroupId(), _user1.getUserId());
+		_userLocalService.addUserGroupUser(
+			userGroup.getUserGroupId(), _user2.getUserId());
+	}
+
+	@Test
+	public void testGetInheritedRoles() throws Exception {
+		CTCollection ctCollection = _ctCollectionLocalService.addCTCollection(
+			null, TestPropsValues.getCompanyId(), TestPropsValues.getUserId(),
+			0, "P1", null);
+
+		MockLiferayResourceResponse mockLiferayResourceResponse =
+			new MockLiferayResourceResponse();
+
+		_mvcResourceCommand.serveResource(
+			_getMockLiferayResourceRequest(
+				_user1.getScreenName(), ctCollection.getCtCollectionId()),
+			mockLiferayResourceResponse);
+
+		JSONArray jsonArray = _getUsersJSONArray(mockLiferayResourceResponse);
+
+		JSONObject itemJSONObject = jsonArray.getJSONObject(0);
+
+		Assert.assertEquals(
+			_user2.getFullName(), itemJSONObject.getString("fullName"));
+	}
+
+	private MockLiferayResourceRequest _getMockLiferayResourceRequest(
+			String userName, long ctCollectionId)
+		throws Exception {
+
+		MockLiferayResourceRequest mockLiferayResourceRequest =
+			new MockLiferayResourceRequest();
+
+		ThemeDisplay themeDisplay = new ThemeDisplay();
+
+		themeDisplay.setCompany(
+			_companyLocalService.getCompany(TestPropsValues.getCompanyId()));
+		themeDisplay.setPermissionChecker(_permissionChecker);
+		themeDisplay.setUser(_user1);
+
+		mockLiferayResourceRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, themeDisplay);
+
+		mockLiferayResourceRequest.setParameter("keywords", userName);
```

Comment:
Why do we need to set the `keywords` param? It's not used in our resource command.

Reply:
I believe it is needed since it is what is being searched [AutocompleteUserMVCResourceCommand.java](https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/AutocompleteUserMVCResourceCommand.java#L80)

Reply:
I guess that is just for the admin user, I can remove it for this test

Reply:
It'll return an empty list if nothing is searched :/

Reply:
Ah ok. We need the keywords to get the users. Otherwise, it will just return an empty list https://github.com/liferay/liferay-portal/blob/master/modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/portlet/action/AutocompleteUserMVCResourceCommand.java#L83